### PR TITLE
[#48] Integrate Firebase with Angular

### LINKED ISSUE #48: Integrate Firebase
<table>
  <tr>
    <th colspan="3">Integrate Firebase with Angular</th>
  </tr>
  <tr>
    <td>Acceptance Test</td>
   <td><span title="Priority">P</span></td> 
   <td><span title="Story Point">SP</span></td>
  </tr>
  <tr>
    <td>
        <ol>

            <li>Firebase works with Angular</li>
        </ol>
    </td>

    <td>1</td>

    <td>3</td>
  </tr>
  <tr>

    <td colspan="3">Add required files for Angular to work with Firebase</td>
  </tr>
</table>

## Task list

- [x] Define Acceptance test

- [x] Assign Story Point Value

- [x] Assign Priority Number

- [x] Integrate Firebase module with Angular

- [x] Verify issue completion with Acceptance Test. Done!

## Related features/user stories

<!--~~~~HELPER~~~~~
Use bullet points like this:
* Related feature [#A]
* Related issue number [#B]
~~~~~~~~~~~~~~~~~~~~-->

* From [#8]

## Additional context
Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by laurent-lao**
Resolve #48 

Warning: This Pull Request also removes previous ASP.NET core files

---

I laid the groundworks for the Firebase to Angular integration.

Do `npm install` (I believe) if you checked out the repo for the first time

* Polling from Firebase database works for now (since the rules are set to public).

* `/user-profile` routes to a Login page for now, but it does not work.

It is currently deployed on the [Firebase web app](https://shutterfeed-soen341.firebaseapp.com)

### CODE REVIEW

**Comment by laurent-lao** [comment]
@tommy-josepovic Noted. I refactored the secrets settings and added a .md guide to get the configs manually from Firebase in 9ee6605

File: angular.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          },
+          "configurations": {
+            "production": {
+              "devServerTarget": "ShutterFeed:serve:production"
+            }
+          }
+        }
+      }
+    }},
+  "defaultProject": "ShutterFeed"
```
Comment (tjosepo) [inline angular.json]: You created this project leaving CSS as default styling, but we could do much more with SCSS.

SCSS is a superset of CSS that uses the same syntax and has a lot more features than CSS, such as nesting, variables, loops, etc.

Here's an example of what you can do in SCSS: 
```scss
.generic-class {
  h1 {
    color: $primary-color;
  }
  
  a {
    font-decoration: none;
    &:hover {
        color: $accent-color;
    }
  }
}
```
Which in vanilla CSS would look like:
```css
.generic-class h1 {
  color: red;
}
  
.generic-class a {
    font-decoration: none;
}
.generic-class  a:hover {
  color: yellow;
}
```

SCSS is can become very useful when working on much larger stylesheets, and the Material library heavily relies on SCSS for styling, so we should switch the project to SCSS simply by appending this after this line:
```javascript
"schematics": {
    "@schematics/angular:component": {
      "styleext": "scss"
    }
  }
```

We would technically also have to rename the .css files into .scss, and rename all references to those files, but we can procrastinate on renaming everything...

File: src/app/user-profile/user-profile.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <!-- template will replace this div -->
+</div>
+
+<!-- User NOT logged in -->
+<ng-template #guest>
+  <h3>Howdy, GUEST</h3>
+  <p>Login to get started...</p>
+
+  <button (click)="auth.googleLogin()">
+    <i class="fa fa-google"></i> Connect Google
```
Comment (tjosepo) [inline src/app/user-profile/user-profile.component.html]: I know this is work in progress, but ``<i>`` tags should never be used.
The point of ``<i>`` tags was to italicize text before CSS was a thing. 
The only reason ``<i>`` still exists is compatibility with older sites.

If you need to contain some text, use ``<span>``. If you want to emphasize some **_important text_**, use ``<em>``. If you want to italicize some _**non-important text**_, use ``<span class="foo">`` and style it using ``foo: { font-style: italic }``.

Also, what does ``fa`` mean in the classname? A more descriptive name would be useful.

**Review by tjosepo — COMMENTED** [review]
It's the first time I really use github to do a code review. Excuse me if this is messy.

In general, good job setting everything up. 🎉

I only have **3 complaints**. 
* **The usage of CSS over SCSS** for the project. This needs to change as SCSS is just a straight upgrade from CSS, and the Material library depends on SCSS for some styling features. An example is provided in the comment above.
* **``<i>`` tags**. Functionally, it doesn't matter, and I feel like everybody will end up hating me for this, but I will try to point out bad practices when I see them. An example is provided in the review comment.
* I don't think I can comment folders, but I think that **instead of having many components called page-something, we can just create a page folder and put all the pages components inside that folder**. This probably doesn't need addressing right now since those components are likely going to get refactored, but I think it's important to let you know that you can place components wherever you want, and nesting components inside of folders them will be necessary in the future if we want to organize our files.

### To do: ###
- [x] Add the code to angular.json to make .scss the default file extension of stylesheets when creating a new component. Optionally rename the already existing .css files and references to those files into .scss.
- [ ] Remove the ``<i>`` tag and use explain/replace fa class name.

File: src/app/user-profile/user-profile.component.html (hunk 2 above)
Comment (laurent-lao) [reply src/app/user-profile/user-profile.component.html]: @tommy-josepovic 
* I agree with SCSS. I'm on it

* See [i tag repurposed as icon tag](https://fontawesome.com/v4.7.0/icon/google)
Note that nothing is stylized by the `<i></i>` tag

* Creating folders per category is very valid. I created those pages to show how to route and fully expect them to be deleted/refactored. (see my [website](https://github.com/laurentlaurent/laurentlao-personal/tree/master/src/app)'s attempt at component organization)

Comment (tjosepo) [reply src/app/user-profile/user-profile.component.html]: I didn't realize it was used for an icon lol.
I think ``<span>`` is considered the best practice, but with Material, icons are going to be ``<mat-icons>`` anyways..

Comment (laurent-lao) [reply src/app/user-profile/user-profile.component.html]: @tommy-josepovic Code Reviewers can also commit changes they feel are necessary (and leave comments to inquire about parts they don't understand) such is the case of the .scss refactoring comment. This is especially the case when it takes less time to edit directly than to explain why an edit is necessary

Comment (tjosepo) [reply src/app/user-profile/user-profile.component.html]: @laurentlaurent We never did that in my internship, but it makes a lot of sense, so I'll do that from now on. Thank you for letting me know!

**Review by tjosepo — APPROVED** [review]

**Pull request merged**